src/Disks/ObjectStorages/Cached/CachedObjectStorage.cpp

```diff
@@ -59,7 +59,7 @@ ReadSettings CachedObjectStorage::patchSettings(const ReadSettings & read_settin
     if (!canUseReadThroughCache())
         modified_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache = true;
 
-    return IObjectStorage::patchSettings(modified_settings);
+    return object_storage->patchSettings(modified_settings);
```

Comment:
So it was a bug before this change?

Reply:
Not really, it did not affect anything, when readObjects is passed to inner object storage (s3) it will apply it's own patch settings for its read.